### ISSUE TITLE
Paginated query over 2i with range return key with wrong index

### ISSUE THREAD

**Issue by peczenyj** [basho/riak#499]
Hi

Performing some queries I find this:

GET 'http://RIAK:8098/buckets/MY_BUCKET/index/expiration_epoch_int/1393462000/1393463030?return_terms=true&max_results=10'

RETURNS [{"1393462839":"FOOBAR"}]

OK 1393462000 <= 1393462839 <= 1393463030

But if I inspect this entry:

GET 'http://RIAK:8098/buckets/MY_BUCKET/keys/FOOBAR
...
x-riak-index-expiration_epoch_int: 1395013128
...

the value of this index id different!

Let me explain what we do: each PUT override the content and the indexes. we update expiration_epoch_int in each PUT ( to delete only old data ). In our configuration we have

allow_mult=false
last_write_wins=true

So... why I query for one range and I find one key in this case?

**Comment by russelldb** [comment]
Did the value of the object change between when you did the 2i query and when you read the object with the get?

**Comment by peczenyj** [comment]
I don't know if it is easy to reproduce. Lets try to give one idea.

We create 14 Gb per day of data stored in my buckets. And we create some 2i indexes ( using elevelDB ) for cleanup and searching. For example, we have one index for last_modified_time to search itens changed in the last 24 hours. When I try to search itens changed between 20 and 21 days ago I find itens changed, in fact, yesterday. But it is not easy to reproduce, I need search and search until find. I note it can happens when we use range queries and pagination and, sometimes, I can receive different results for the same query when I run in sequence.

**Comment by peczenyj** [comment]
No, the value of the object does not change between the 2i query and when I read the object. In this particular case, if it is true I expect some value +120 days from now. 1395013128 is the epoch for '2014-03-16T23:38:48'

**Comment by peczenyj** [comment]
One more interesting fact: IF I perform the same query over and over, I can find different (wrong) results. But when I inspect one entry, If I perform the same query again and again I can find this result. It is like: when we GET the data, it change what is returning the value with a wrong secondary index value ( like clean some cache ). It make sense?

**Comment by jaredmorrow** [comment]
/cc @engelsanchez 

What version of Riak are you seeing this on @peczenyj?
